ui/packages/consul-ui/app/components/code-editor/layout.scss

```diff
@@ -32,3 +33,121 @@
 %code-editor > pre {
   display: none;
 }
+
+%code-editor .ember-basic-dropdown-trigger {
+  background-color: var(--tone-gray-900);
+  color: var(--black);
+  border-radius: 2px;
+  border: 1px solid;
+  border-color: var(--tone-gray-700);
+  margin: 0 8px;
+  width: 120px;
+  height: 32px;
+  display: flex;
+  align-items: center;
+  flex-direction: row;
+}
+
+.toolbar {
+  background-color: rgb(var(--tone-gray-050));
+  border: 1px solid rgb(var(--tone-gray-100));
+  border-bottom-color: rgb(var(--tone-gray-600));
+  border-top-color: rgb(var(--tone-gray-400));
+  position: relative;
+  margin-top: 4px;
+
+  &::after {
+    background: linear-gradient(to right, rgb(var(--tone-gray-600)), rgba(var(--tone-gray-600), 0));
+    bottom: 0;
+    content: '';
+    position: absolute;
+    right: 0;
+    top: 0;
+    width: 4px;
+    z-index: 2;
+  }
+
+  .input,
+  .select {
+    min-width: 190px;
+  }
+}
+
+.toolbar-label {
+  padding: 8px;
+  color: rgb(var(--tone-gray-800));
+}
+
+.toolbar-scroller {
+  align-items: center;
+  display: flex;
+  height: 44px;
+  justify-content: space-between;
+  overflow-x: auto;
+  width: 100%;
+
+  &::-webkit-scrollbar {
+    border-bottom: 1px solid rgb(var(--tone-gray-400));
+    height: 4px;
+  }
+
+  &::-webkit-scrollbar-thumb {
+    background: rgb(var(--tone-gray-400));
+    border-radius: 4px;
+  }
+}
+
+.toolbar-filters,
+.toolbar-actions {
+  align-items: center;
+  display: flex;
+  flex: 1;
+  white-space: nowrap;
+  justify-content: space-between;
+
+  .title {
+    color: rgb(var(--tone-gray-900));
+    display: inline-block;
+    font-size: 14px;
+    font-weight: 700;
+    padding: 0 8px;
+  }
+  .actions {
+    display: flex;
+    flex-direction: row;
+    margin: 0 10px;
+    align-items: center;
+
+    .copy-button {
+      margin-left: 10px;
+    }
+  }
+}
+
+.toolbar-link {
+  border: 0;
+  color: var(--black);
+  transition: background-color 150ms;
+
+  &:hover {
+    background-color: rgb(var(--tone-gray-050));
+    border: 0;
+    color: rgb(var(--tone-blue-500));
+  }
+
+  &:active {
+    box-shadow: none;
+  }
+
+  &.popup-menu-trigger {
+    height: 2.5rem;
+    padding: 0.5rem 0.857rem;
+  }
+}
+
+.toolbar-separator {
+  border-right: 1px solid rgb(var(--tone-gray-300));
+  height: 32px;
+  margin: 0 4px;
+  width: 0;
+}
```

Comment:
Feels like there a lot here, so I'm gonna take a deeper look, so up to you if you want to make changes now or wait til then. but generally:

1. If you could use that CSS props for decoration things like borders etc that I linked to above that'd be great.
2. Use a tone instead of black (as above)
3. All in all this will need splitting into `layout` and `skin` files - but we can leave that until absolute last thing as its just cleanup/consistency - I'd rather get everything finished and then do a final clean up, but up to you if you wanted to do it now instead.

Reply:
I agree and I like the splitting, right, since I am going to refactor the Toolbar anyway now that I know how to do it ;) 
I will let you know when this is ready for review again so that I don't waste too much of your time.

Reply:
Sounds great 👍